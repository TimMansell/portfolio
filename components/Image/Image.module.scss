.container {
  display: flex;
  justify-content: center;
}

.image {
  max-width: 100%;
}

.isPadded {
  padding: 0 var(--vr);
}
