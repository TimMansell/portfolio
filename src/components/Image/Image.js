import React from 'react';
import PropTypes from 'prop-types';

export const Image = ({ src, alt, width, height }) => {
  const img = require(`../${src}`);

  return (
    <img
      src={img}
      alt={alt}
      width={width}
      height={height}
      title={alt}
      loading="lazy"
    />
  );
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  width: PropTypes.string,
  height: PropTypes.string,
};

export default Image;
