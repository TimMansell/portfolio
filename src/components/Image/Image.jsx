import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './Image.module.scss';

export const Image = ({ src, alt, width, height, isPadded }) => {
  const img = new URL(`../${src}`, import.meta.url).href;

  const containerClasses = classnames(styles.container, {
    [styles.isPadded]: isPadded,
  });

  return (
    <div className={containerClasses}>
      <img
        className={styles.image}
        src={img}
        alt={alt}
        width={width}
        height={height}
        title={alt}
        loading="lazy"
      />
    </div>
  );
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  width: PropTypes.string,
  height: PropTypes.string,
  isPadded: PropTypes.bool,
};

export default Image;
