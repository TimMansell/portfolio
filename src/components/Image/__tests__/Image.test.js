import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Image from '../Image';

const props = {
  src: 'Image/__tests__/img/test.jpg',
  alt: 'alt text',
  width: '50',
};

describe('Image', () => {
  it('should render my component', () => {
    render(<Image {...props} />);
  });

  it('should match snapshot', () => {
    const { asFragment } = render(<Image {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
