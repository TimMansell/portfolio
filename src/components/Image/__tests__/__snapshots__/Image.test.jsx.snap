// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > should match snapshot 1`] = `
<div
  className="container"
>
  <img
    alt="alt text"
    className="image"
    loading="lazy"
    src="file:///H:/projects/portfolio/src/components/Image/__tests__/img/test.jpg"
    title="alt text"
    width="50"
  />
</div>
`;
